src/loader-recorder-v2.ts

```diff
@@ -200,6 +211,40 @@ async function getRequestPerformanceEntry(
     return performanceEntry
 }
 
+function _tryReadXHRBody(body: Document | XMLHttpRequestBodyInit | null | undefined): string | null {
+    if (_isString(body)) {
+        return body
+    }
+
+    if (_isDocument(body)) {
+        return body.textContent
+    }
+
+    if (_isFormData(body)) {
+        return _formDataToQuery(body)
+    }
+
+    return null
+}
+
+/**
+ * According to MDN https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/response
+ * @param response an ArrayBuffer, a Blob, a Document, a JavaScript object, or a string, depending on the value of XMLHttpRequest.responseType, that contains the response entity body.
+ */
+function _tryReadXHRResponseBody(response: any): string | null {
```

Comment:
in each of these `_tryReadX` methods we can ignore body types we don't want to process e.g. `ArrayBuffer`